libp2p/transports/tortransport.nim

```diff
@@ -0,0 +1,141 @@
+
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import std/[oids, sequtils]
+import chronos, chronicles, strutils
+import stew/byteutils
+import ../multicodec
+import transport,
+      tcptransport,
+      ../wire,
+      ../stream/[lpstream, connection, chronosstream],
+      ../multiaddress,
+      ../upgrademngrs/upgrade
+
+const
+  Socks5TransportTrackerName* = "libp2p.tortransport"
+
+type
+  TorTransport* = ref object of Transport
+    transportAddress: TransportAddress
+    tcpTransport: TcpTransport
+
+proc new*(
+  T: typedesc[TorTransport],
+  address: string, 
+  port: Port): T {.public, raises: [Defect, TransportAddressError]} =
```

Comment:
You can directly take a `TransportAddress` here, which allows user to give a unix socket if he wants to (and you don't have to worry about parsing errors)

Reply:
Done